### PR TITLE
Changed ec2 connection method

### PR DESCRIPTION
**Pull request by wimnat**
Pull request to resolve issue https://github.com/ansible/ansible/issues/12959

### CODE REVIEW

**Comment by willthames** [comment]
I'm not 100% sure why this change works, as I thought `ec2_connect` was just an special case of `connect_to_aws('boto.ec2',...)` (see https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/ec2.py#L202-L213)

However, if `ec2_connect` is this broken, should we look at removing it completely?

**Comment by wimnat** [comment]
I do feel the option of connect_to_aws and ec2_connect is unnecessary.  I would prefer to just use the connect_to_aws method all the time in all modules.

File: cloud/amazon/ec2.py [hunk 1]
```diff
@@ -1353,10 +1353,13 @@ def main():
     if not HAS_BOTO:
         module.fail_json(msg='boto required for this module')
 
-    ec2 = ec2_connect(module)
-
     region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module)
 
+    try:
+        ec2 = connect_to_aws(boto.ec2, region, **aws_connect_params)
```
Comment (willthames) [inline cloud/amazon/ec2.py:1359]: Note that the previous line returns `aws_connect_kwargs`

You're effectively passing it a dict from somewhere else

**Comment by willthames** [comment]
I think we should fix whatever is going wrong in `ec2_connect`. `ec2_connect` eliminates a lot of boilerplate (it handles the `get_aws_connection_info`, it does exception handling, it even handles the `ec2_url` which most alternatives do not). 

I suspect better security_token handling in `get_aws_connection_info` might get closer to a genuine fix.

**Comment by wimnat** [comment]
Hmmm I'll close this for now but it warrants more research.  You're right, it shouldn't fix it but i've seen a similar approach fix other connection problems too.

**Comment by willthames** [comment]
I just wonder if throwing all the `aws_connect_params` away makes it work (as that's really the only difference - you're passing a completely different dict, although one that must exist, and therefore be populated with reasonable values)

**Pull request closed without merging**